cudax/test/event/event_smoke.cu

```diff
@@ -0,0 +1,129 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of CUDA Experimental in CUDA C++ Core Libraries,
+// under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES.
+//
+//===----------------------------------------------------------------------===//
+
+#include <cuda/experimental/event.cuh>
+
+#include "../common/utility.cuh"
+#include "../hierarchy/testing_common.cuh"
+#include <catch2/catch.hpp>
+
+namespace
+{
+namespace test
+{
+cudax::event_ref fn_takes_event_ref(cudax::event_ref ref)
+{
+  return ref;
+}
+} // namespace test
+} // namespace
+
+static_assert(!_CUDA_VSTD::is_default_constructible_v<cudax::event_ref>);
+static_assert(!_CUDA_VSTD::is_default_constructible_v<cudax::event>);
+static_assert(!_CUDA_VSTD::is_default_constructible_v<cudax::timed_event>);
+
+TEST_CASE("can construct an event_ref from a cudaEvent_t", "[event]")
+{
+  ::cudaEvent_t ev;
+  CUDAX_REQUIRE(::cudaEventCreate(&ev) == ::cudaSuccess);
+  cudax::event_ref ref(ev);
+  CUDAX_REQUIRE(ref.get() == ev);
+  CUDAX_REQUIRE(!!ref);
+  // test implicit converstion from cudaEvent_t:
+  cudax::event_ref ref2 = ::test::fn_takes_event_ref(ev);
```

Comment:
I don't know if its better, but when I have a pattern of thing, thing2, thing3 in sections starting with a comment, I started to put them into catch2 SECTION("name") { thing...} instead. This way the comment ends up being displayed on error and there is no need to number things since these are different scopes